docs/src/darray.md

```diff
@@ -0,0 +1,135 @@
+ 0.12689    0.194842   0.263186   0.213071   0.535613    0.246888      0.579931   0.699231   0.441449    0.882772
+ 0.916144   0.21305    0.629293   0.329303   0.299889    0.127453      0.644012   0.311241   0.713782    0.0554386
+ ⋮                                                       ⋮          ⋱
+ 0.430369   0.597251   0.552528   0.795223   0.46431     0.777119      0.189266   0.499178   0.715808    0.797629
+ 0.235668   0.902973   0.786537   0.951402   0.768312    0.633666      0.724196   0.866373   0.0679498   0.255039
+ 0.605097   0.301349   0.758283   0.681568   0.677913    0.51507    …  0.654614   0.37841    0.86399     0.583924
+ 0.824216   0.62188    0.369671   0.725758   0.735141    0.183666      0.0401394  0.522191   0.849429    0.839651
+ 0.578047   0.775035   0.704695   0.203515   0.00267523  0.869083      0.0975535  0.824887   0.00787017  0.920944
+ 0.805897   0.0275489  0.175715   0.135956   0.389958    0.856349      0.974141   0.586308   0.59695     0.906727
+ 0.212875   0.509612   0.85531    0.266659   0.0695836   0.0551129     0.788085   0.401581   0.948216    0.00242077
+ 0.512997   0.134833   0.895968   0.996953   0.422192    0.991526   …  0.838781   0.141053   0.747722    0.84489
+ 0.283221   0.995152   0.61636    0.75955    0.072718    0.691665      0.151339   0.295759   0.795476    0.203072
+ 0.0946639  0.496832   0.551496   0.848571   0.151074    0.625696      0.673817   0.273958   0.177998    0.563221
+ 0.0900806  0.127274   0.394169   0.140403   0.232985    0.460306      0.536441   0.200297   0.970311    0.0292218
+ 0.0698985  0.463532   0.934776   0.448393   0.606287    0.552196      0.883694   0.212222   0.888415    0.941097
+```
+
+An existing array can be distributed by using `Distribute`.
+```julia
+julia> Z = zeros(100, 500);
+
+julia> Dzeros = Distribute(Blocks(10, 50), Z)
+Distribute{Float64, 2}(100, 500)
+
+julia> compute(Dzeros)
+Dagger.DArray{Any, 2, typeof(cat)}(100, 500)
+```
+
+## Using Distributed Arrays
+
+Arithmetic using `DArray` and related types are done lazily. See the following example.
+
+```julia
+julia> using Distributed; addprocs(6)
+6-element Vector{Int64}:
+ 2
+ 3
+ 4
+ 5
+ 6
+ 7
+
+julia> using Dagger
+
+julia> DX = rand(Blocks(50,50), 100, 100)
+Dagger.AllocateArray{Float64, 2}(100, 100)
+
+julia> DY = DX + DX
+Dagger.BCast{Base.Broadcast.Broadcasted{Dagger.DaggerBroadcastStyle, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}, typeof(+), Tuple{Dagger.AllocateArray{Float64, 2}, Dagger.AllocateArray{Float64, 2}}}, Float64, 2}(100, 100)
+
+julia> DZ = DY .* 3
+Dagger.BCast{Base.Broadcast.Broadcasted{Dagger.DaggerBroadcastStyle, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}, typeof(*), Tuple{Dagger.BCast{Base.Broadcast.Broadcasted{Dagger.DaggerBroadcastStyle, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}, typeof(+), Tuple{Dagger.AllocateArray{Float64, 2}, Dagger.AllocateArray{Float64, 2}}}, Float64, 2}, Int64}}, Float64, 2}(100, 100)
+
```

Comment:
This would be a good time to point out how the types start getting longer as we start effectively nesting operations.

Reply:
This behavior will go away by the next release, so I'll skip doing this.